lib/checkclass.cpp

```diff
@@ -2559,7 +2559,8 @@ const std::list<const Token *> & CheckClass::getVirtualFunctionCalls(const Funct
         const Function * callFunction = tok->function();
         if (!callFunction ||
             function.nestedIn != callFunction->nestedIn ||
-            (tok->previous() && tok->previous()->str() == "."))
+            Token::simpleMatch(tok->previous(), ".") ||
+            !(tok->astParent() && (tok->astParent()->str() == "(" || (tok->astParent()->str() == "::") && Token::simpleMatch(tok->astParent()->astParent(), "("))))
```

Comment:
tok->astParent() && (tok->astParent()->str() == "(" || (tok->astParent()->str() == "::")
=>
Token::Match(tok->astParent(), "(|::")

Reply:
Hmm, there is a parenthesis between the two conditions.

On a side note, this still needs approval: https://github.com/danmar/cppcheck/pull/3871

Reply:
Not quite what I meant to write, thanks.